include/boost/variant/variant.hpp

```diff
@@ -2452,16 +2455,16 @@ struct make_variant_over
 private: // precondition assertions
 
     BOOST_STATIC_ASSERT(( ::boost::mpl::is_sequence<Types>::value ));
+    typedef typename mpl::copy<Types, boost::mpl::front_inserter<mpl::list<> > >::type copied_sequence_t;
```

Comment:
This will reverse the sequence. Many people use `boost::get` function with indexes (not with types) and such change in behavior will break their code. Also add a test case for that case (make sure that after `typedef boost::make_variant_over<boost::mpl::joint_view<v1::types, v2::types>::type>::type v3;` the expression `get<0>` returns an `int`)

Try using something like this:

```
typedef typename boost::mpl::insert_range<
      boost::mpl::list<>
    , boost::mpl::end< boost::mpl::list<> >::type
    , Types
    >::type copied_sequence_t;
```